src/main/java/Duke/Parser.java

```diff
@@ -0,0 +1,146 @@
+                }
+                time = scanner.nextLine();
+                LocalDate d1 = LocalDate.parse(time.substring(1));
+                Task task = new Event(description, d1);
+                tasklist.addTask(task);
+                int total = tasklist.size();
+                bot.addTask(total,task);
+            }
+            else if (str.equals("list")) {
+                bot.printTasks(tasklist.getOldTasks(),tasklist.getNewTasks());
+                scanner.nextLine();
+            }
+            else if (str.equals("unmark")) {
+                String strnum = scanner.next();
+                int num = Integer.valueOf(strnum);
+                Task task;
+                if (num > tasklist.oldTasksSize()) {;
+                    task = tasklist.getNewTasks(num-tasklist.oldTasksSize()-1);
+                    task.markUndone();
+                    tasklist.setNewTasks(num-tasklist.oldTasksSize()-1,task);
+                } else {
+                    task = tasklist.getOldTasks(num-1);
+                    task.markUndone();;
+                    tasklist.setOldTasks(num-1,task);
+                }
+                bot.markTask(false);
+                System.out.println(task);
+                scanner.nextLine();
+            } else if (str.equals("mark")) {
+                String strnum = scanner.next();
+                int num = Integer.valueOf(strnum);
+                Task task;
+                if (num > tasklist.oldTasksSize()) {
+                    task = tasklist.getNewTasks(num-tasklist.oldTasksSize()-1);
+                    task.markDone();
+                    tasklist.setNewTasks(num-tasklist.oldTasksSize()-1, task);
+                } else {
+                    task = tasklist.getOldTasks(num-1);
+                    task.markDone();
+                    tasklist.setOldTasks(num-1, task);
+                }
+                bot.markTask(true);
+                System.out.println(task);
+                scanner.nextLine();
+            }
+            else{
+                if (tasklist.oldTasksSize() < initialSize) {
+                    storage.replaceTasks("data/Duke2.txt", tasklist.getOldTasks(), tasklist.getNewTasks());
+                } else {
+                    storage.replaceTasks("data/Duke2.txt", tasklist.getOldTasks(), tasklist.getNewTasks());
+                    //storage.addTasks("data/Duke2.txt",tasklist.newTasks);
+                }
+                bot.displayError();
+            }
+
+        }  while (!str.equals("bye"));
+        if (str.equals("bye")) {
+            if(tasklist.oldTasksSize() < initialSize){
+                storage.replaceTasks("data/Duke2.txt", tasklist.getOldTasks(), tasklist.getNewTasks());
+            } else{
+                storage.replaceTasks("data/Duke2.txt", tasklist.getOldTasks(), tasklist.getNewTasks());
+                //storage.addTasks("data/Duke2.txt",tasklist.newTasks);
+            }
+            bot.goodBye();
+        }
+   }

```

Comment:
Perhaps it is better to make multiple shorter methods for easier readability and future reference?

Reply:
oh ya changed it alr thanks!!